src/extension.ts

```diff
@@ -1,9 +1,40 @@
 import { ExtensionContext } from "vscode";
 import { GodotTools } from "./godot-tools";
-
+import * as path from "path";
+import * as fs from "fs"
+import * as vscode from 'vscode'
 
 let tools: GodotTools = null;
 
+export class DocContent {
+	static docItems: vscode.QuickPickItem[] = []
+	static builtIns = new Set()
+	static dataDirectory
+
+	constructor(dataPath : string) {
+		DocContent.dataDirectory = path.join(dataPath, 'godot', 'doc')
+		let indexLocation = path.join(dataPath, 'godot', 'doc', `index.json`)
+
+		fs.readFile(indexLocation, 'utf8', function (err, data) {
```

Comment:
Use `fs.readFileSync` to keep the workflow more simple and clear.
To make new contributors form C++ feel good.